### PR TITLE
Error upgrades

### PR DESCRIPTION
**Pull request by DXCanas**
Getting error upgrades into develop. See #4051, #4054, #4047

@christianmemije I had to rework some of your code with vuex3 modules. Please test/revise :pray: 

@ralphiee22 Had to resolve some merge conflicts in your code. Please test/revise :pray: 

@jonboiser Just a quick lookover to see if the old work with state were translated properly?

### CODE REVIEW

**Comment by DXCanas** [comment]
After I've gotten an approval, I'll squash my commits

File: kolibri/plugins/facility_management/assets/src/views/UserPage/UserCreateModal.vue
Comment (jonboiser) [inline kolibri/plugins/facility_management/assets/src/views/UserPage/UserCreateModal.vue]: handleApiError should be a core action. did you mean

```
...mapActions(['handleApiError']) // note no namespace
```

File: kolibri/core/assets/src/styles/core-theme.scss
Comment (christianmemije) [inline kolibri/core/assets/src/styles/core-theme.scss:23]: We might want to just get rid of this as a global variable.

File: kolibri/core/assets/src/views/AppBody.vue
Comment (christianmemije) [inline kolibri/core/assets/src/views/AppBody.vue:57]: We can get rid of this computed prop by just referencing `windowIsSmall` instead.

File: kolibri/core/assets/src/views/AppError/index.vue
Comment (christianmemije) [inline kolibri/core/assets/src/views/AppError/index.vue:1]: Let's delete the ErrorBox component.

File: kolibri/core/assets/src/views/AppError/ReportErrorModal.vue
Comment (christianmemije) [inline kolibri/core/assets/src/views/AppError/ReportErrorModal.vue]: `as a .txt file`?

File: kolibri/plugins/user/assets/src/modules/profile/actions.js
Comment (christianmemije) [inline kolibri/plugins/user/assets/src/modules/profile/actions.js:47]: Getting `unknown local mutation type: SET_PROFILE_ERROR, global type: profile/SET_PROFILE_ERROR`

File: kolibri/plugins/facility_management/assets/src/views/UserPage/EditUserModal.vue
Comment (christianmemije) [inline kolibri/plugins/facility_management/assets/src/views/UserPage/EditUserModal.vue:212]: Getting `unknown local action type: setError, global type: userManagement/setError`

**Review by christianmemije — COMMENTED** [review]
Front end code looks good. Left a few comments on issues I encountered while testing the error constant handling.

File: kolibri/plugins/user/assets/src/modules/profile/actions.js
Comment (jonboiser) [reply kolibri/plugins/user/assets/src/modules/profile/actions.js:47]: That must be coming from line 42. Should that be a different mutation?

Comment (christianmemije) [reply kolibri/plugins/user/assets/src/modules/profile/actions.js:47]: SET_PROFILE_ERROR**S** (MIssing the S)

File: kolibri/core/assets/src/views/AppError/index.vue
Comment (DXCanas) [reply kolibri/core/assets/src/views/AppError/index.vue:1]: Ah. A bunch of files I'd deleted came back in the merge. Good catch.

File: kolibri/core/assets/src/views/AppError/ReportErrorModal.vue
Comment (DXCanas) [reply kolibri/core/assets/src/views/AppError/ReportErrorModal.vue]: Weird discrepancy between screencap and doc strings @khangmach 
https://docs.google.com/document/d/1X-p6Vnj3aSnQFC8lF_ZZo7mIvnMAwNZBbruPbL3FrOc/edit

Comment (khangmach) [reply kolibri/core/assets/src/views/AppError/ReportErrorModal.vue]: Looks like I forgot the "a". Christian's suggestion works too.

**Comment by christianmemije** [comment]
Need to prettify ./kolibri/core/assets/test/state/store.spec.js

**Comment by DXCanas** [comment]
@christianmemije @jonboiser Addressed all comments. Once tests pass and I get an approval, gonna squash and merge!

File: kolibri/core/auth/serializers.py
Comment (jamalex) [inline kolibri/core/auth/serializers.py:44]: Since the user-facing string is no longer being translated here, I'm assuming this one is just for debugging, and it's now specified (and translated) in the frontend?

File: kolibri/core/utils.py
Comment (jamalex) [inline kolibri/core/utils.py:35]: A more explicit mapping rather than just uppercasing what comes out of DRF might be safer, but we can revisit that later.

**Review by jamalex — APPROVED** [review]
Don't see any issues with the Python code (assuming it all works -- didn't do any testing).

**Comment by jamalex** [comment]
(Note: Didn't review any of the non-Python code)

File: kolibri/core/assets/src/styles/core-theme.scss
Comment (indirectlylit) [reply kolibri/core/assets/src/styles/core-theme.scss:23]: agreed

**Pull request merged**
